src/topology/pcm.c

```diff
@@ -929,6 +929,15 @@ int tplg_parse_pcm(snd_tplg_t *tplg, snd_config_t *cfg,
 			continue;
 		}
 
+		if (strcmp(id, "ignore_suspend") == 0) {
+			err = parse_flag(n,
+				SND_SOC_TPLG_LNK_FLGBIT_VOICE_WAKEUP,
+				&pcm->flag_mask, &pcm->flags);
+			if (err < 0)
+				return err;
+			continue;
+		}
+
```

Comment:
this is very confusing. Existing usages of 'ignore_suspend' are limited to BE dailinks, typically dmic used for hotwording.

I don't really see how this definition is extended to the FE and PCM levels? This isn't really necessary since the ignore_suspend set in the BE will keep all paths connected to that BE active.

Put differently, we didn't need this extension before to enable hotwording usages, so why is this needed now?
What does this bring?

@ranj063 FYI.

Reply:
@amadeuszslawinski-intel @crojewsk-intel any update here ? still needed ?

Reply:
@plbossart @lgirdwood Excuse my delay. In regard to code adding `ignore_suspend` flag parsing, the idea was simply to add missing parses while addressing other issues. If there is no parser for given field, it's basically a redundant information - user has no means of notifying kernel about it. So, I suggest we either add the missing parsers or, remove the redundant fields from the topology description.

Reply:
Friendly ping @lgirdwood @plbossart.

Reply:
My feedback was not really addressed. WoV was enabled without any topology support for ignore_suspend, what makes it a requirement now?

best to deal with @ranj063 on this.

Reply:
Patch adding the serialization fixes the missing link between userspace and the kernel space. I'm not sure whether we use it or not for the WoV scenarios, but it's an orthogonal subject.

If the property cannot be propagated to the kernel, then it should not be part of the topology interface. So, if we do not want the add the parsing, then the field should be gone - it's dead in its current form.

Reply:
I think there is some unnecessary confusion because of how flag is named. Sadly because it is UAPI, we can't easily change it...

While flag itself is named SND_SOC_TPLG_LNK_FLGBIT_VOICE_WAKEUP it is used to change ignore_suspend bit:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/sound/soc/soc-topology.c#n1700
in my opinion better name would be SND_SOC_TPLG_LNK_FLGBIT_IGNORE_SUSPEND.

It is defined in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/sound/asoc.h#n170 and can be set in 2 structures: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/sound/asoc.h#n530
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/sound/asoc.h#n556

As @crojewsk-intel mentioned it is orthogonal to WoV support, however as far as WoV support goes we use it to tell driver that FE or BE has pipelines which will be active during S0idle state to support WoV during suspend.

Reply:
Is feedback above enough to finalize this PR?

Reply:
Friendly reminder @lgirdwood @plbossart @perexg.